File: lambdas/algorithms/searching/java-searching/src/main/java/model/SearchResult.java
package model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class SearchResult {
    // metadata
    private String timestamp;
    private long timestampRaw;

    // input params
    private String array;
    private String target;
    private String algorithm;
    private String language;
    private String memorySize;
    private String region;

    // metrics
    private long iterations;
    private long comparisons;
    private double timeTaken;
    private double cost;

    // results
    private String executionStatus;
    private String searchStatus;
    private int foundIndex;
}

File: lambdas/algorithms/sorting/python-sorting/src/models/search_result.py
from dataclasses import dataclass
from dataclasses_json import dataclass_json

@dataclass_json
@dataclass
class SearchResult:
    timestamp: str
    timestampRaw: int
    array: str
    target: str
    algorithm: str
    language: str
    memorySize: str
    region: str
    iterations: int
    comparisons: int
    timeTaken: float
    cost: float
    executionStatus: str
    searchStatus: str
    foundIndex: int